src/reconciliation.js

```diff
@@ -221,19 +221,29 @@ export function patchAccount(
     next.balance = BigNumber(updatedRaw.balance);
     changed = true;
   }
-
-  if (updatedRaw.balanceHistory) {
-    const { week } = updatedRaw.balanceHistory;
+  const { balanceHistory } = updatedRaw;
+  if (balanceHistory) {
+    const { week } = balanceHistory;
     const accountWeek = account.balanceHistory && account.balanceHistory.week;
-    const refreshBalanceHistory =
-      !week || // there is no week yet
-      !accountWeek || // there were no balance history yet
-      // last datapoint changes will force a refresh
-      week[week.length - 1][1] !==
-        accountWeek[accountWeek.length - 1].value.toString();
-    if (refreshBalanceHistory) {
-      next.balanceHistory = fromBalanceHistoryRawMap(updatedRaw.balanceHistory);
-      changed = true;
+    if (week && accountWeek) {
```

Comment:
if there is no week or account week, we still should reconciliate to "empty" the object, otherwise you can be stuck when we disable the data.

I suggest to make a function "shouldRefreshBalance" history and to return true in these case. Also please try to keep previous comments.

Reply:
🤷‍♂ sure

Reply:
Solved in current code